modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCacheStatsHolder.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cache.common.tier;
+
+import org.opensearch.common.cache.stats.DefaultCacheStatsHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * A tier-aware version of DefaultCacheStatsHolder. Overrides the incrementer functions, as we cannot just add the on-heap
+ * and disk stats to get a total for the cache as a whole. For example, if the heap tier has 5 misses and the disk tier
+ * has 4, the total cache has had 4 misses, not 9. The same goes for evictions. Other stats values add normally.
+ * This means for misses and evictions, if we are incrementing for the on-heap tier, we have to increment only the leaf nodes
+ * corresponding to the on-heap tier itself, and not its ancestors.
+ */
```

Comment:
This should be the case only if the disk tier is enabled? Else, the miss on on-heap tier should also be considered for total hits/miss?

Reply:
Yes, this is the case in the code, will update the comment to reflect it